crypto/fipsmodule/modes/gcm.c

```diff
@@ -155,7 +155,29 @@ static size_t hw_gcm_encrypt(const uint8_t *in, uint8_t *out, size_t len,
   if (!len_blocks) {
     return 0;
   }
-  aes_gcm_enc_kernel(in, len_blocks * 8, out, Xi, ivec, key);
+
+  if (CRYPTO_is_ARMv8_GCM_8x_capable() && len >= 192) {
+    switch(key->rounds) {
+    case 10:
+      aesv8_gcm_8x_enc_128(in, len_blocks * 8, out, Xi, ivec, key);
+      break;
+    case 12:
+      aesv8_gcm_8x_enc_192(in, len_blocks * 8, out, Xi, ivec, key);
+      break;
+    case 14:
+      aesv8_gcm_8x_enc_256(in, len_blocks * 8, out, Xi, ivec, key);
+      break;
+    default:
+      // This function is allowed to process no input;
+      // the subsequent logic after calling it can process
+      // the entire input.
```

Comment:
not sure I understand the comment?

Reply:
I changed it. I can remove it if it doesn't make much sense.